iroh-net/src/hp/derp/http/client.rs

```diff
@@ -37,46 +37,67 @@ const CONNECT_TIMEOUT: Duration = Duration::from_secs(10);
 const MESH_CLIENT_REDIAL_DELAY: Duration = Duration::from_secs(5);
 
 #[derive(Debug, thiserror::Error)]
+/// Possible connection errors on the [Client]
 pub enum ClientError {
```

Comment:
I am sorry, but could we unify these to be in the order 1. docs, 2. derive/macros, 3. definition?

```rust
/// Possible connection errors on the [Client]
#[derive(Debug, thiserror::Error)]
pub enum ClientError {
```